pkg/chroottar/chroottar.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2015 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package chroottar
+
+import (
+	"archive/tar"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strconv"
+	"syscall"
+
+	"github.com/coreos/rkt/pkg/procflavor"
+	ptar "github.com/coreos/rkt/pkg/tar"
+)
+
+const (
+	flavor       = "extracttarchroot"
+	fileMapFdNum = 3
+)
+
+func init() {
+	procflavor.Add(flavor, extractTar)
+}
+
+func extractTar() error {
+	runtime.LockOSThread()
```

Comment:
@yifan-gu right. As chroot should be per process, the need to lock a goroutine to a specific thread shouldn't be needed. @vcaputo any thoughts?